### PR TITLE
[Feature] Support video demo

### PR DESCRIPTION
**Pull request by irvingzhang0512**
## Motivation

Video Demo, use local video file or webcam as input, visualize predictions by imshow or generating local video file.

## Use cases (Optional)

PS: Not sure if I should upload a demo video `demo/demo.mp4`.

```shell
python demo/video_demo.py ${VIDEO_FILE} ${CONFIG_FILE} ${CHECKPOINT_FILE} [--device ${DEVICE_NAME}] [--palette-thr ${PALETTE}] \
    [--show] [--show-wait-time {SHOW_WAIT_TIME}] [--output-file {OUTPUT_FILE}] [--output-fps {OUTPUT_FPS}] \
    [--output-height {OUTPUT_HEIGHT}] [--output-width {OUTPUT_WIDTH}] [--opacity {OPACITY}]
```

Examples:

```shell
python demo/video_demo.py demo/demo.mp4 configs/cgnet/cgnet_680x680_60k_cityscapes.py \
    checkpoints/cgnet_680x680_60k_cityscapes_20201101_110253-4c0b2f2d.pth \
    --device cuda:0 --palette cityscapes --show
```

### CODE REVIEW

**Comment by Junjun2016** [comment]
Hi @irvingzhang0512 
Thanks for your contribution, nice PR!

> PS: Not sure if I should upload a demo video demo/demo.mp4.

You could provide a link for us to download the `demo.mp4`,
We will download the file and upload it to OSS.
So users can download it from OSS and inference the demo.

File: docs/get_started.md [hunk 1]
```diff
@@ -216,3 +216,19 @@ python demo/image_demo.py demo/demo.jpg configs/pspnet/pspnet_r50-d8_512x1024_40
...
+Now we also provide a demo script to test a single video.
+
+```shell
+python demo/video_demo.py ${VIDEO_FILE} ${CONFIG_FILE} ${CHECKPOINT_FILE} [--device ${DEVICE_NAME}] [--palette-thr ${PALETTE}] \
+    [--show] [--show-wait-time {SHOW_WAIT_TIME}] [--output-file {OUTPUT_FILE}] [--output-fps {OUTPUT_FPS}] \
+    [--output-height {OUTPUT_HEIGHT}] [--output-width {OUTPUT_WIDTH}] [--opacity {OPACITY}]
+```
+
+Examples:
+
```
Comment (Junjun2016) [inline docs/get_started.md:229]: Could download the demo through OSS first and put it to` demo/`.

**Comment by Junjun2016** [comment]
@irvingzhang0512 
If the video is short, we could save it in the repository like https://github.com/open-mmlab/mmtracking/blob/master/demo/demo.mp4.

**Comment by MengzhangLI** [comment]
Hi, @irvingzhang0512 thanks for your nice pr!

Could you please upload demo.mp4 or just send this model to my e-mail (mcmong@pku.edu.cn)?

Best,

**Comment by irvingzhang0512** [comment]
Sorry for the late response, I'm a little busy right now and I'll take care of this PR this weekend.

**Comment by Junjun2016** [comment]
> Sorry for the late response, I'm a little busy right now and I'll take care of this PR this weekend.

Thanks a lot.

**Comment by RockeyCoss** [comment]
Please merge the master branch into your branch, thank you.

**Comment by Junjun2016** [comment]
Hi @irvingzhang0512 
Could you please upload the video demo if your time is available?

**Comment by MengzhangLI** [comment]
Hi, @irvingzhang0512 

Any progress? We can review this pr in the next few days.

Thanks.

**Comment by MengzhangLI** [comment]
Hi, sorry to bother you. @irvingzhang0512 

Anything to update?

**Comment by irvingzhang0512** [comment]
Sorry ,  I have worked overtime for a few weeks... I'll take care of this tonight...

**Comment by irvingzhang0512** [comment]
BTW, I'm not familiar with image semgentation public dataset, is there any video suitable this demo?

**Comment by MengzhangLI** [comment]
Hoping these links may be useful to you ;)
They are all about Cityscapes dataset: 

Original videos https://www.cityscapes-dataset.com/examples/#videos
Examples from youtube: 
https://www.youtube.com/results?search_query=cityscapes+dataset
https://www.youtube.com/watch?v=lGOjchGdVQs

Best,

**Comment by irvingzhang0512** [comment]
@MengzhangLI 

This PR is ready for review.

Create a 10s long demo video based on Cityscapes test set berlin folder. Maybe we can upload this video to your server and add `wget xxx`  in getting started docs.

https://user-images.githubusercontent.com/22089207/144212749-44411ef4-b564-4b37-96d4-04bedec629ab.mp4

**Comment by MengzhangLI** [comment]
> 

Hi, please fix the lint error. If our codebase is your upstream, you may

use `git pull upstream master`  and `pre-commit run --all-files`.

File: docs/get_started.md [hunk 2]
```diff
@@ -9,24 +9,24 @@
...
+|         0.14.1         | mmcv-full>=1.3.7, <1.4.0  |
+|         0.14.0         | mmcv-full>=1.3.1, <1.3.2  |
+|         0.13.0         | mmcv-full>=1.3.1, <1.3.2  |
+|         0.12.0         | mmcv-full>=1.1.4, <1.3.2  |
+|         0.11.0         | mmcv-full>=1.1.4, <1.3.0  |
+|         0.10.0         | mmcv-full>=1.1.4, <1.3.0  |
+|         0.9.0          | mmcv-full>=1.1.4, <1.3.0  |
+|         0.8.0          | mmcv-full>=1.1.4, <1.2.0  |
+|         0.7.0          | mmcv-full>=1.1.2, <1.2.0  |
+|         0.6.0          | mmcv-full>=1.1.2, <1.2.0  |
```
Comment (MengzhangLI) [inline docs/get_started.md:29]: Could you keep the typesetting of this part the same with current version?

**Comment by MengzhangLI** [comment]
It looks good to me! Thanks for your nice pr.

image

Please fix lint version, typesetting and sentences in `get_started.md` above. Thanks!

Best,

File: docs/get_started.md [hunk 3]
```diff
@@ -216,3 +216,20 @@ python demo/image_demo.py demo/demo.jpg configs/pspnet/pspnet_r50-d8_512x1024_40
 ```
 
 A notebook demo can be found in [demo/inference_demo.ipynb](../demo/inference_demo.ipynb).
+
+Now we also provide a demo script to test a single video.
+
+```shell
```
Comment (MengzhangLI) [inline docs/get_started.md:222]: Add  `wget xxx` content in this shell, thank you.

**Review by MengzhangLI — APPROVED** [review]

**Review by xvjiarui — APPROVED** [review]

**Pull request merged**